invenio_records_resources/resources/files/resource.py

```diff
@@ -174,6 +206,35 @@ def read_content(self):
 
         return item.send_file(), 200
 
+    @request_view_args
+    def read_archive(self):
+        """Read a zipped version of all files."""
+        all_in_one_enabled = current_app.config.get("ALL_IN_ONE_DOWNLOAD_ENABLED", True)
+        if not all_in_one_enabled:
+            abort(404)
+
+        id_ = resource_requestctx.view_args["pid_value"]
+        files = self.service.list_files(
+            g.identity,
+            id_,
+        )
+
+        if files is None:
+            abort(404)
+
+        entries = files.to_dict()["entries"]
+        contents = {}
+        for entry in entries:
+            file = self.service.get_file_content(g.identity, id_, entry["key"])
+            contents[entry["key"]] = file
```

Comment:
minor: seeing this part now, it feels like there's a mix of service/presentation layers (see [data flow architecture docs](https://inveniordm.docs.cern.ch/develop/architecture/software/)). I could imagine having all the service calls for listing and opening the files inside one "bigger" service call on the `FileService` (also for efficiency).